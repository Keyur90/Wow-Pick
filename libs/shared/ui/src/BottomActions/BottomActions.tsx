import React, { ReactNode } from 'react';
import { makeStyles } from '@material-ui/core/styles';
import { Theme } from '@material-ui/core';
import AppBar from '@material-ui/core/AppBar';
import { Container } from '../Container';

interface BottomActionsPropTypes {
  children: NonNullable<ReactNode>;
  isPrimary?: boolean;
}

const useStyles = makeStyles<Theme, Omit<BottomActionsPropTypes, 'children'>>((theme) => ({
  appBar: {
    top: 'auto',
    bottom: ({ isPrimary }) => theme.spacing(isPrimary ? 0 : 8),
    background: 'none',
    padding: theme.spacing(2, 0),
    boxShadow: 'none',
  },
  content: {
    display: 'grid',
    gridAutoFlow: 'row',
    gridRowGap: theme.spacing(2),
  },
}));

const BottomActions: React.FC<BottomActionsPropTypes> = React.memo(({ children, isPrimary = true }) => {
  const classes = useStyles({ isPrimary });

  return (
    <AppBar position="fixed" color="primary" className={classes.appBar}>
      <Container>
        <div className={classes.content}>{children}</div>
      </Container>
    </AppBar>
  );
});

export { BottomActions };
