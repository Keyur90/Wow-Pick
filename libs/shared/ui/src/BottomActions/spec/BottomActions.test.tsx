import React from 'react';
import { render } from '@testing-library/react';
import { BottomActions } from '../BottomActions';

test('renders BottomActions', () => {
  const { container } = render(<BottomActions>Some bottom actions</BottomActions>, {
    container: document.body,
  });

  expect(container).toMatchSnapshot();
});
