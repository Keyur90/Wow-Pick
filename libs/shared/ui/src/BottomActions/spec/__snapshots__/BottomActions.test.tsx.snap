// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders BottomActions 1`] = `
<body>
  <header
    class="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary makeStyles-appBar-1 makeStyles-appBar-3 mui-fixed MuiPaper-elevation4"
  >
    <div
      class="MuiContainer-root makeStyles-root-4 MuiContainer-maxWidthLg"
    >
      <div
        class="makeStyles-content-2"
      >
        Some bottom actions
      </div>
    </div>
  </header>
</body>
`;
